src/main/scala/piconot/PicoBotAPI.scala

```diff
@@ -0,0 +1,114 @@
+// Internal DSL I think
+
+import picolib._
+import picolib.maze._
+import picolib.semantics._
+import java.io.File
+import scala.util.parsing.combinator._
+
+
+trait Line
+// class Lines extends Line 
+
+
+def MakeDirection(direction: String): MoveDirection = {
+  var returnedDirection = direction match {
+    case "north" => North 
+    case "North" => North
+    case  "N" => North
+    case "n" => North
+    case "south" => South
+    case  "South" => South 
+    case "S" => South 
+    case "s" => South
+    case "east" => East
+    case "East" => East 
+    case "E" => East 
+    case "e" => East
+    case "west" => West 
+    case "West" => West 
+    case "W" => West 
+    case "w" => West
+  }
+  return returnedDirection
+}
+
+def MakeSurroundings(north: String, east: String, west: String, south: String): Surroundings = {
+  var returnedNorth = north match {
+    case "north" => Blocked
+    case "North" => Blocked 
+    case "N" => Blocked
+    case "n" => Blocked
+    case "*" => Anything
+    case "Any" => Anything
+    case "any" => Anything
+    case "x" => Open
+    case "X" => Open
+    case "Empty" => Open
+    case "empty" => Open
+  }
+
+  var returnedSouth = south match {
+    case "south" => Blocked
+    case "South" => Blocked
+    case "S" => Blocked
+    case "s" => Blocked
+    case "*" => Anything 
+    case "Any" => Anything
+    case "any" => Anything
+    case "x" => Open
+    case "X" => Open 
+    case "Empty" => Open
+    case "empty" => Open
+  }
+
+  var returnedEast = east match {
+    case "east" => Blocked
+    case "East" => Blocked 
+    case "E" => Blocked 
+    case "e" => Blocked
+    case "*" => Anything 
+    case "Any" => Anything 
+    case "any" => Anything
+    case "x" => Open 
+    case "X" => Open 
+    case "Empty" => Open 
+    case "empty" => Open
+  }
```

Comment:
I'm a fan of the cases that you included that allows for a user write their rules the way that they want.